pdfs/src/Pdf/FromLatex.hs

```diff
@@ -0,0 +1,206 @@
+
+module Pdf.FromLatex where
+
+import qualified Data.ByteString as BS
+import Path.IO
+import Path
+import System.Directory
+import System.IO.Temp
+import System.Process.Typed
+
+escapeInLatex :: Text -> Text
+escapeInLatex text = text
+
+type Locale = Text
+
+data Date = Date
+    { year  :: Int
+    , month :: Int
+    , day   :: Int
+    }
+
+dateToString :: Locale -> Date -> Text
+dateToString _ (Date y m d) = show d <> "." <> show m <> "." <> show y
+
+data StudentInfo = StudentInfo
+    { siName        :: Text
+    , siDateOfBirth :: Date
+    , siSections    :: [Section]
+    }
+
+data Section = Section
+    { sEducator :: Text
+    , sDiploma  :: Diploma
+    , sCourses  :: [Course]
+    }
+
+data Diploma = Diploma
+    { dEducatorUrl     :: Text
+    , dEducationPeriod :: (Int, Int)
+    , dDiplomaId       :: Text
+    , dDegreeLevel     :: Text
+    , dMajor           :: Text
+    , dMinor           :: Text
+    , dPartTime        :: Bool
+    }
+
+data Course = Course
+    { cName        :: Text
+    , cLanguage    :: Text
+    , cHours       :: Int
+    , cEstcCredits :: Maybe Int
+    , cResult      :: Text
+    }
+
+newtype ResourcePath = ResourcePath { unResourcePath :: FilePath }
+
+produce :: Locale -> StudentInfo -> ResourcePath -> IO ByteString
+produce loc info (ResourcePath resources) = do
+    withSystemTempDirectory "faircv" $ \dir -> do
+        resPath <- parseRelDir resources
+        tmpPath <- parseAbsDir dir
+        copyDirRecur resPath tmpPath
+
+        let input = encodeUtf8 $ generate loc info
+        withCurrentDirectory dir $ do
+            let action =
+                    runProcess
+                    $ setStdin (byteStringInput input)
+                    $ setStdout byteStringOutput
+                    $ "xelatex"
```

Comment:
I believe that `xelatex` should be added to `shell-native.nix`/`default.nix` in order to ensure its presence in the system. Adding it to the code deployed in the cluster might be trickier, or maybe just adding it to `default.nix` would suffice. Please, consult @kirelagin on this matter.